event.go

```diff
@@ -67,10 +70,21 @@ func (ed *DefaultEventHandler) Post(height uint64, msg *types.EventData) error {
 	}
 	// TODO: this is stupid, fix it
 	eventData := EventData(*msg)
-	ed.stash.add(height, &eventData)
+	ed.eventCache = append(ed.eventCache, &eventData)
 	return nil
 }
 
+func (ed *DefaultEventHandler) Commit(height uint64) {
+	for _, eventData := range ed.eventCache {
+		ed.stash.add(height, eventData)
```

Comment:
Seeing as we now add batches of events to the stash rather than one event at a time it'd be good to update the stash interface to accept an array of events to reduce locking.